Making task logs interactive.

16yuki0702 opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes
Related issue is https://github.com/tektoncd/cli/issues/414

adding interactive menu to task logs.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Regenerate the manpages and docs with `make docs` and `make man` if needed.
- [ ] Run the code checkers with `make check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

<!--
Does your PR contain User facing changes?

If so, briefly describe them here so we can include this description in the
release notes for the next release!
-->

```
task logs will be interactive
```

danielhelfand commented:
/ok-to-test

pradeepitm12 commented:
/test pull-tekton-cli-unit-tests

danielhelfand commented on pkg/cmd/task/logs.go:
```diff
@@ -0,0 +1,196 @@
+	opts := options.NewLogOptions(p)
+
+	eg := `
+  # interactive mode: shows logs of the selected task run
```
```suggestion
  # interactive mode: shows logs of the selected taskrun
```

danielhelfand commented on pkg/cmd/task/logs.go:
```diff
@@ -0,0 +1,196 @@
+  # interactive mode: shows logs of the selected taskrun of the given task
+    tkn task logs task -n namespace
+
+  # show logs of given task for last run
```
```suggestion
  # show logs of given task for last taskrun
```

danielhelfand commented on pkg/cmd/task/logs.go:
```diff
@@ -0,0 +1,196 @@
+    tkn task logs task -n namespace --last
+
+  # show logs for given task and taskrun
+    tkn task logs task run -n namespace
```
```suggestion
    tkn task logs taskrun -n namespace
```

danielhelfand replied:
I'm assuming that you went with this because of what is in pipeline logs, so feel free to change it there as well. Otherwise, we can get it in a follow up pr.

danielhelfand commented on pkg/cmd/task/logs_test.go:
```diff
@@ -0,0 +1,519 @@
+		},
+		{
+			name:      "Specify task name and limit as 2",
+			limit:     2,
```
Would this test be better with `limit: 1` to assess whether limit returns less than the full amount?

danielhelfand commented on pkg/cmd/task/logs.go:
```diff
@@ -0,0 +1,196 @@
+		DisableFlagsInUseLine: true,
+		Short:                 "Show task logs",
+		Example:               eg,
+		SilenceUsage:          true,
```
I wonder if this is needed anymore since `root.go` has `SilenceUsage: True`, but no need to dig further in this pr. It might be something we should look into.

16yuki0702 commented:
@danielhelfand 
Thank you for review!!
I tried to fix the resource name to 'taskrun' only task files in this PR for now (not include pipeline files) :pray:
and remove unnecessary option :pray:

danielhelfand commented:
> @danielhelfand
> Thank you for review!!
> I tried to fix the resource name to 'taskrun' only task files in this PR for now (not include pipeline files) 🙏
> and remove unnecessary option 🙏

I think you actually removed `DisableFlagsInUseLine: true,`. Sorry to do this, but let's keep `DisableFlagsInUseLine: true,` and `SilenceUsage: true` for now. I'll open an issue for this, and we could go about removing across the CLI.

16yuki0702 commented:
@danielhelfand Oops... I'm so sorry why I deleted ```DisableFlagsInUseLine: true``` is just a my big mistake and misunderstanding :innocent:
 
> I think you actually removed DisableFlagsInUseLine: true,. Sorry to do this, but let's keep DisableFlagsInUseLine: true, and SilenceUsage: true for now. I'll open an issue for this, and we could go about removing across the CLI.

But I got it, I added the ```DisableFlagsInUseLine: true``` again for now :pray:

16yuki0702 commented:
@danielhelfand 
> @16yuki0702 Thanks! Can you please squash your commits?

Thanks!! squashed :thumbsup:
